src/ResourceManager/MachineLearning/Commands.MachineLearning/Cmdlets/GetAzureMLWebService.cs

```diff
@@ -48,25 +49,27 @@ protected override void RunCmdlet()
             }
             else
             {
-                IList<WebService> services;
-                if (!string.IsNullOrWhiteSpace(this.ResourceGroupName))
+                // This is a collection of web services get call, so determine which flavor it is
+                Func<Task<ResponseWithContinuation<WebService[]>>> getFirstServicesPageCall = () => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(null, this.CancellationToken);
+                Func<string, Task<ResponseWithContinuation<WebService[]>>> getNextPageCall = nextLink => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(nextLink, this.CancellationToken);
+                if (this.ResourceGroupName != null)
                 {
-                    services = this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAndGroupAsync(
-                                                        this.ResourceGroupName,
-                                                        null,
-                                                        this.CancellationToken).Result;
-                }
-                else
-                {
-                    services = this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(
-                                                        null,
-                                                        this.CancellationToken).Result;
+                    // This is a call for resource retrieval within a resource group
+                    getFirstServicesPageCall = () => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAndGroupAsync(this.ResourceGroupName, null, this.CancellationToken);
+                    getNextPageCall = nextLink => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAndGroupAsync(this.ResourceGroupName, nextLink, this.CancellationToken);
```

Comment:
@NonStatic2014 it looks like if `this.ResourceGroupName` isn't null, then these two methods will be called twice (once before the if-statement, and once inside. Is there a reason for doing this instead of doing the following:

```cs
if (this.ResourceGroupName != null)
{
     Func<Task<ResponseWithContinuation<WebService[]>>> getFirstServicesPageCall = () => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(this.ResourceGroupName, null, this.CancellationToken);
     Func<string, Task<ResponseWithContinuation<WebService[]>>> getNextPageCall = nextLink => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(this.ResourceGroupName, nextLink, this.CancellationToken);
}
else
{
     Func<Task<ResponseWithContinuation<WebService[]>>> getFirstServicesPageCall = () => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(null, this.CancellationToken);
     Func<string, Task<ResponseWithContinuation<WebService[]>>> getNextPageCall = nextLink => this.WebServicesClient.GetAzureMlWebServicesBySubscriptionAsync(nextLink, this.CancellationToken);
}
```